tsdb/db.go

```diff
@@ -232,6 +239,10 @@ type DB struct {
 	// Cancel a running compaction when a shutdown is initiated.
 	compactCancel context.CancelFunc
 
+	// compactStart helps shift the compactions start time.
+	// TODO: better name, especially given the `start` variable in Compact().
```

Comment:
any suggestions? I can't think of anything else than `compactShiftStart` :)

(is it ok to add a field here? we could pass a pointer or sth around `Compact`)